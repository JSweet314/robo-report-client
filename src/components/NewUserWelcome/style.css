.new-user-welcome {
  width: 50%;
  min-width: 320px;
  margin: auto;
  padding: 1rem;
  font-size: 2rem;
  justify-content: center;
  align-items: center;
  text-align: center;
}

.new-user-welcome button {
  display: block;
  margin: auto;
  width: 50%;
  min-width: 4rem;
  font-size: 2rem;
}
