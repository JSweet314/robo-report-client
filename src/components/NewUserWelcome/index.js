import React from 'react';
import PropTypes from 'prop-types';
import './style.css';

const NewUserWelcome = ({ toggleWelcome }) => {
  return (
    <article className='new-user-welcome'>
      <h2>Welcome aboard!</h2>
      <p>
        In order to make reporting unwanted robocalls to the FCC easier, we
        need to collect some information that is included with every report.
      </p>
      <p>
        Required information can include legal name, address, phone number
        (of course), type of phone, etc.
      </p>
      <button onClick={() => toggleWelcome()}>Get Started!</button>
    </article>
  );
};

NewUserWelcome.propTypes = {
  toggleWelcome: PropTypes.func.isRequired
};

export default NewUserWelcome;