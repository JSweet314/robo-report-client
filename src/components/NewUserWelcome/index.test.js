import React from 'react';
import { shallow } from 'enzyme';
import NewUserWelcome from './index';

describe('NewUserWelcome', () => {
  let wrapper;
  const mockToggleWelcome = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<NewUserWelcome toggleWelcome={mockToggleWelcome}/>);
  });

  it('should match a snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call toggleWelcome when the get started button is pressed', () => {
    const button = wrapper.find('button');
    button.simulate('click');
    expect(mockToggleWelcome).toHaveBeenCalled();
  });
});