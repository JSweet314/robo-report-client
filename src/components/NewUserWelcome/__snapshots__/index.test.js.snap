// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewUserWelcome should match a snapshot 1`] = `
<article
  className="new-user-welcome"
>
  <h2>
    Welcome aboard!
  </h2>
  <p>
    In order to make reporting unwanted robocalls to the FCC easier, we need to collect some information that is included with every report.
  </p>
  <p>
    Required information can include legal name, address, phone number (of course), type of phone, etc.
  </p>
  <button
    onClick={[Function]}
  >
    Get Started!
  </button>
</article>
`;
